### ISSUE TITLE
Rule Change: no-implicit-globals should not report exported

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#16327]
### What rule do you want to change?

no-implicit-globals

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* exported testFunc */

/**
 * @returns {int}
 */
function testFunc() {
    "use strict";

    return 1 + 2;
}
```

### What does the rule currently do for this code?

When using with lexicalBindings `/*eslint no-implicit-globals: ["error", {"lexicalBindings": true}]*/` it gives an error

### What will the rule do after it's changed?

Not report an error, if the variable/function is "exported"

### Additional comments

This was already requested previously but closed due to age https://github.com/eslint/eslint/issues/10066

Essentially it should have the same behavior as rule `no-unused-vars`, where exported variables/functions don't count as unused

**Comment by mdjermanovic** [comment]
> Not report an error, if the variable/function is "exported"

Makes sense to me :+1: 

I think the main purpose of this rule is to prevent accidental creation of global variables. Declaring `/* exported testFunc */` indicates the intention to create a global `testFunc`, and is an alternative to the style `window.testFunc = function (){}` that this rule currently enforces.

**Comment by nzakas** [comment]
Makes sense to me. Should this be a bug rather than an enhancement?

**Comment by sosukesuzuki** [comment]
Can I work on this?

**Comment by mdjermanovic** [comment]
> Should this be a bug rather than an enhancement?

I think this was an oversight in the design because the `/* exported */` comments feature already existed when this rule was created. Currently allowed way is to add a `/* global testFunc: writable */` comment (here's a [playground example](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWltcGxpY2l0LWdsb2JhbHM6IFtcImVycm9yXCJdICovXG5cbi8qIGdsb2JhbCB0ZXN0RnVuYzogd3JpdGFibGUgKi9cblxuZnVuY3Rpb24gdGVzdEZ1bmMoKSB7XG4gICAgXCJ1c2Ugc3RyaWN0XCI7XG5cbiAgICByZXR1cm4gMSArIDI7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6eyJlczYiOnRydWV9fX0=)), which seems wrong because the variable doesn't exist before running the script.

**Comment by nzakas** [comment]
Just to clarify: you’re saying this is an enhancement because the rule intentionally didn’t consider this behavior?

**Comment by mdjermanovic** [comment]
Sorry for the confusion. Supporting `/* global testFunc: writable */` as a way to allow global `function testFunc` by using eslint directives may imply that the other directives such as `/* exported testFunc */` were considered and intentionally not supported, but it's hard to tell because `exported` was not explicitly mentioned anywhere in the original issue and the PR.

Either way, I think we can treat this as a bug fix.

However, if this requires adding a new property on Variable objects (https://github.com/eslint/eslint/pull/16343#discussion_r978961061), then it seems more appropriate to mark the change as a `feat`, as it will include an addition to the core.

**Comment by nzakas** [comment]
> However, if this requires adding a new property on Variable objects (https://github.com/eslint/eslint/pull/16343#discussion_r978961061), then it seems more appropriate to mark the change as a feat, as it will include an addition to the core.

Sounds good.
